seaworthy/dockerhelper.py

```diff
@@ -74,38 +72,91 @@ def teardown(self):
             self.remove_container(container)
         self._container_ids = None
 
-        # Remove the network
-        if self._network is not None:
-            self._network.remove()
-            self._network = None
+    def _teardown_networks(self):
+        # Remove the default network
+        if self._default_network is not None:
+            self._default_network.remove()
+            self._default_network = None
 
-        # We need to close the underlying APIClient explicitly to avoid
-        # ResourceWarnings from unclosed HTTP connections.
-        self._client.api.close()
-        self._client = None
+        # Remove all other networks
+        for network_id in self._network_ids.copy():
+            # Check if the network exists before trying to remove it
+            try:
+                network = self._client.networks.get(network_id)
+            except docker.errors.NotFound:
+                continue
+
+            log.warning("Network '{}' still existed during teardown".format(
+                network.name))
+
+            self.remove_network(network)
+        self._network_ids = None
+
+    def get_default_network(self, create=True):
+        """
+        Get the default bridge network that containers are connected to if no
+        other network options are specified.
+
+        :param create:
+            Whether or not to create the network if it doesn't already exist.
+        """
+        if self._default_network is None and create:
+            log.debug("Creating default network...")
+            self._default_network = self.create_network(
+                'default', driver='bridge')
+
+        return self._default_network
 
     def create_container(self, name, image, **kwargs):
         container_name = self._resource_name(name)
         log.info("Creating container '{}'...".format(container_name))
+
+        network = self._get_container_network(kwargs)
+        network_id = network.id if network is not None else None
+
         container = self._client.containers.create(
-            image, name=container_name, detach=True, network=self._network.id,
+            image, name=container_name, detach=True, network=network_id,
             **kwargs)
 
+        if network is not None:
+            self._connect_container_network(container, network, aliases=[name])
+
+        # Keep a reference to created containers to make sure they are cleaned
+        # up
+        self._container_ids.add(container.id)
+
+        return container
+
+    def _get_container_network(self, create_kwargs):
+        # If a network is specified use that
+        network = create_kwargs.pop('network', None)
```

Comment:
Not sure how terrible this is...

Reply:
I'm generally not enthusiastic about modifying mutable args, but I don't think it's a major issue here and I can't think of a cleaner way to do this offhand.